metricbeat/module/docker/docker.go

```diff
@@ -50,6 +50,8 @@ func FetchStats(client *docker.Client) ([]DockerStat, error) {
 
 	containersList := []DockerStat{}
 	for _, container := range containers {
+		// This is currently very inefficient as docker calculates the average for each request,
+		// means each request will take at least 1s. Getting all stats at once is implemented here: https://github.com/docker/docker/pull/25361
 		containersList = append(containersList, exportContainerStats(client, &container))
```

Comment:
Each request takes at least 2 seconds: https://github.com/docker/docker/blob/master/cli/command/container/stats_helpers.go#L148 I'm planing to open an issue in Docker to discuss if there could be flag that raw flags are shipped. There is also https://github.com/docker/docker/pull/25361 which will be useful to fetch all stats for the containers at once.